app/src/main/java/com/g11x/checklistapp/ChecklistItemActivity.java

```diff
@@ -18,35 +18,57 @@
 package com.g11x.checklistapp;
 
 import android.content.Intent;
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import android.view.Menu;
-import android.view.MenuInflater;
-import android.view.MenuItem;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
 import android.widget.TextView;
 import android.widget.ToggleButton;
 
 import com.g11x.checklistapp.data.ChecklistItem;
 import com.g11x.checklistapp.data.ChecklistManager;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
 
 public class ChecklistItemActivity extends AppCompatActivity {
 
   private ChecklistItem checklistItem;
   private ToggleButton isDone;
+  private DatabaseReference databaseRef;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_checklist_item);
 
-    int index = this.getIntent().getIntExtra("index", 0);
-    checklistItem = ChecklistManager.get(getApplicationContext()).getItems().get(index);
+    databaseRef = FirebaseDatabase.getInstance().getReferenceFromUrl(
```

Comment:
There are cases where `onStart` fires, but not `onCreate` (I think anyway, [the activity lifecycle](https://developer.android.com/training/basics/activity-lifecycle/starting.html) is complicated). I don't think it's an issue with the app currently, but in general it's better to pull args out of the intent in `onStart`, so how about moving this section to `onStart`?